zipkin/src/main/java/zipkin/internal/Collector.java

```diff
@@ -55,13 +63,35 @@ protected void acceptSpans(byte[] serializedSpans, D decoder, Callback<Void> cal
     List<S> spans;
     try {
       spans = decodeList(decoder, serializedSpans);
+      spans = filterSpans(spans, callback);
     } catch (RuntimeException e) {
       callback.onError(errorReading(e));
       return;
     }
     accept(spans, callback);
   }
 
+  /**
+   * Take the list of spans and pump them through pre-configured filters
+   *
+   * @param spans
+   */
+  List<S> filterSpans(List<S> spans, Callback<Void> callback) {
+    // These should never be processed in parallel.
+    // If you need parallel processing put that into your specific filter implementation.
+    boolean isV1Span = spanType.getType().equals(Span.class);
```

Comment:
nit: One day we may need a `SpanVersion` enum. In that case, all spans could hypothetically derive from `VersionedSpan`, which would obviate the need for this kind of logic. You could simply switch on `VersionedSpan.getVersion()`.